cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hdfs/AbstractHDFSStats.java

```diff
@@ -39,4 +63,85 @@
   public String getServiceName() {
     return SERVICE_NAME;
   }
+
+  public String getWebURL() {
+    try {
+      if (HAUtil.isHAEnabled(conf, getNameService())) {
+        URL haWebURL = getHAWebURL();
+        if (haWebURL != null) {
+          return haWebURL.toString();
+        }
+      } else {
+        try (FileSystem fs = FileSystem.get(conf)) {
+          URL webUrl = rpcToHttpAddress(fs.getUri());
+          if (webUrl != null) {
+            return webUrl.toString();
+          }
+        }
+      }
+      lastCollectFailed = false;
+    } catch (Exception e) {
+      // TODO: remove once CDAP-7887 is fixed
+      if (!lastCollectFailed) {
+        LOG.warn("Error in determining HDFS URL. Web URL of HDFS will not be available in HDFS operational stats.", e);
+      }
+      lastCollectFailed = true;
+    }
+    return null;
```

Comment:
You should annotate the method with `@Nullable` if this method can return null. Also add javadoc to tell what does it mean when `null` is returned.